src/main/java/org/javarosa/core/model/data/GeoPointData.java

```diff
@@ -190,8 +192,19 @@ public String toString() {
         return getDisplayText();
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (o == null || getClass() != o.getClass()) return false;
```

Comment:
You could add a reference equality check as the first step:
`if (this == o) return true;`